### PR TITLE
Invoke on-failure for exception thrown in transaction

### PR DESCRIPTION
**Pull request by nodename**
#8

### CODE REVIEW

File: src/com/degel/re_frame_firebase/database.cljs [hunk 1]
```diff
@@ -55,24 +55,27 @@
...
 (defn- transactioner [{:keys [path transaction-update on-success on-failure apply-locally]}]
-  (.transaction (fb-ref path)
-                (transaction-update-wrapper transaction-update)
-                (success-failure-wrapper on-success on-failure)
-                ;; Force apply-locally to be a boolean, as required by .transaction.
-                (if (or (false? apply-locally)
-                        (nil? apply-locally))
-                  false
-                  true)))
+  (try
```
Comment (pdelfino) [inline src/com/degel/re_frame_firebase/database.cljs:58]: @nodename 
Could you add a docstring?

File: src/com/degel/re_frame_firebase/database.cljs [hunk 2]
```diff
@@ -55,24 +55,27 @@
...
+                          (nil? apply-locally))
+                    false
+                    true))
+    (catch :default e (on-failure e))))
+
 
 (def transaction-effect transactioner)
 
 (defn- swapper [{:keys [path f argv on-success on-failure apply-locally]}]
   (transactioner
```
Comment (pdelfino) [inline src/com/degel/re_frame_firebase/database.cljs:73]: @nodename 
Could you add a docstring on this function too?

**Comment by pdelfino** [comment]
Nice, @nodename. I know the functions were not written by you. But, since you understood them to do a tweak, could you add a docstring to these 2 functions?

It is great reading a codebase with functions documented 😄

**Review by jojoucsd — APPROVED** [review]

**Review by pdelfino — APPROVED** [review]

File: src/com/degel/re_frame_firebase/database.cljs (hunk 2 above)
Comment (nodename) [reply src/com/degel/re_frame_firebase/database.cljs:73]: I'm not going to add docstrings. I know nothing aobut these functions other than what you can understand from their names.

Comment (pdelfino) [reply src/com/degel/re_frame_firebase/database.cljs:73]: alright then

**Pull request merged**